File: src/main/java/jp/co/sunarch/apps/idprovider/model/oauth2/password/ResourceOwnerPasswordCredentialsAccessTokenRequest.java
package jp.co.sunarch.apps.idprovider.model.oauth2.password;

import jp.co.sunarch.apps.idprovider.model.oauth2.Scope;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

/**
 * アクセストークンリクエスト
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ResourceOwnerPasswordCredentialsAccessTokenRequest {

  private String username;
  private String password;
  private Set<Scope> scope;

}

File: src/main/kotlin/jp/co/sunarch/apps/idprovider/model/oauth2/password/ResourceOwnerPasswordCredentialsAccessTokenRequest.kt
package jp.co.sunarch.apps.idprovider.model.oauth2.password

import jp.co.sunarch.apps.idprovider.model.oauth2.Scope
import lombok.AllArgsConstructor
import lombok.Builder
import lombok.Data
import lombok.NoArgsConstructor

/**
 * アクセストークンリクエスト
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
class ResourceOwnerPasswordCredentialsAccessTokenRequest {

  private val username: String? = null
  private val password: String? = null
  private val scope: Set<Scope>? = null

}
